src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadMeControl.xaml.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.VisualStudio.Markdown.Platform;
+using NuGet.PackageManagement.UI.ViewModels;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGet.PackageManagement.UI
+{
+
+    /// <summary>
+    /// Interaction logic for PackageMetadataReadMeControl.xaml
+    /// </summary>
+    public partial class PackageMetadataReadMeControl : UserControl, IDisposable
+    {
+        public static readonly DependencyProperty PackageMetadataProperty =
+            DependencyProperty.Register(
+                nameof(PackageMetadata),
+                typeof(DetailedPackageMetadata),
+                typeof(PackageMetadataReadMeControl),
+                new PropertyMetadata(OnPropertyChanged));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        private IMarkdownPreview _markdownPreview;
+#pragma warning restore CS0618 // Type or member is obsolete
+        private bool _disposed = false;
+
+        private ReadMePreviewViewModel ReadMeViewModel { get => (ReadMePreviewViewModel)DataContext; }
+
+        [Obsolete]
+        public PackageMetadataReadMeControl()
+        {
+            InitializeComponent();
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        public DetailedPackageMetadata PackageMetadata
+        {
+            get
+            {
+                return (DetailedPackageMetadata)GetValue(PackageMetadataProperty);
+            }
+            set
+            {
+                UpdateControl(PackageMetadata, value);
+                SetValue(PackageMetadataProperty, value);
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _markdownPreview?.Dispose();
+                var viewModel = (ReadMePreviewViewModel)DataContext;
+                viewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
+            }
+
+            _disposed = true;
+        }
+
+        private static void OnPropertyChanged(
+           DependencyObject dependencyObject,
+           DependencyPropertyChangedEventArgs e)
+        {
+            var control = (PackageMetadataReadMeControl)dependencyObject;
```

Comment:
This will throw if it's used for anything other than a ReadMe control. 
Alternatively, could rename it to be specialized, like  `PackageMetadataPropertyChanged`.
```suggestion
            var control = dependencyObject as PackageMetadataReadMeControl;
```